src/IntegratorLock/IntegratorLockCleaner.php

```diff
@@ -41,25 +46,31 @@ public function deleteLock(): void
     {
         $lockFilePath = $this->config->getIntegratorLockFilePath();
 
-        unlink($lockFilePath);
-
-        if (!$this->isLockFileChangeTrackedByGit($lockFilePath)) {
+        if ($this->isLockFileIgnoredByGit($lockFilePath) && !file_exists($lockFilePath)) {
             return;
         }
+        if (file_exists($lockFilePath)) {
+            unlink($lockFilePath);
+        }
+        $gitignorePath = $this->config->getProjectRootDirectory() . static::GITIGNORE_FILE;
+
+        if (!file_exists($gitignorePath) || strpos((string)file_get_contents($gitignorePath), $this->config::INTEGRATOR_LOCK) === false) {
+            file_put_contents($gitignorePath, $this->config::INTEGRATOR_LOCK . PHP_EOL, FILE_APPEND);
```

Comment:
Please use Filesystem so this code could be testable. The same for the `unlink()`. Btw, in that case `file_exists` could be dropped.